bump compose version to v2.24.0 (#19089)
docker/docs: content/compose/release-notes.md

For more detailed information, see the [release notes in the Compose repo](https://github.com/docker/compose/releases/).

## 2.24.0
{{< release-date date="2024-01-11" >}}

### Update

> - Dependencies upgrade: bump cli to 25.0.0-beta.3
> - Dependencies upgrade: bump compose-go to 2.0.0-beta.3
> - Dependencies upgrade: bump ddev to 1.22.4 
> - Dependencies upgrade: bump golang to 1.21.6 
> - Dependencies upgrade: bump golangci-lint to 1.55.2

Reviewer: We don't need to mention `ddev` and `golangci-lint` which are used on CI
```suggestion
- Dependencies upgrade: bump cli to 25.0.0-beta.3
- Dependencies upgrade: bump compose-go to 2.0.0-beta.3
- Dependencies upgrade: bump golang to 1.21.6 
```

Author: removed in https://github.com/docker/docs/commit/4bdc7942d3aa6d5a8e9cc18c6019098908d09472

### Bug fixes and enhancements
- Introduce `docker compose attach` to attach local standard input, output, and error streams to a service's running container
- Introduce `docker compose stats` to display a live stream of container(s) resource usage statistics
- Introduce `docker compose ps --orphans` to include/exclude services not declared
- Introduce `docker compose logs --index` to select a replica container 
- Introduce `docker compose build --with-dependencies` to also build dependencies 
- Add source policies for build
- Include disabled services for shell completion
- Restore `Project` in ps json output
- Add OCI 1.0 fallback support for AWS ECR
- Build do not require environment to be resolved
- Send out a cancel event on SIGINT/SIGTERM signal for `compose up`
- Fix log by exposing services ports when `--verbose`

> - Fix configs are mounted under /\<id\>

Author: Thanks for the suggestions! Could you please review the latest version?

> ### Bug fixes and enhancements
> - Introduce `docker compose attach` to attach local standard input, output, and error streams to a service's running container
> - Introduce `docker compose stats` to display a live stream of container(s) resource usage statistics
> - Introduce `docker compose ps --orphans` to include/exclude services not declared
> - Introduce `docker compose logs --index` to select a replica container 
> - Introduce `docker compose build --with-dependencies` to also build dependencies 
> - Add source policies for build
> - Include disabled services for shell completion
> - Restore `Project` in ps json output
> - Add OCI 1.0 fallback support for AWS ECR
> - Build do not require environment to be resolved
> - Send out a cancel event on SIGINT/SIGTERM signal for `compose up`
> - Fix log by exposing services ports when `--verbose`
> - Fix configs are mounted under /\<id\>
> - Fix combination of `--pull always --no-build`
> - Fix race condition in log printer
> - Fix `docker compose up` teardown when command context is cancelled

Reviewer: ```suggestion
### Bug fixes and enhancements
- Introduced `docker compose attach` to attach local standard input, output, and error streams to a service's running container
- Introduced `docker compose stats` to display a live stream of container(s) resource usage statistics
- Introduced `docker compose ps --orphans` to include/exclude services not declared
- Introduced `docker compose logs --index` to select a replica container 
- Introduced `docker compose build --with-dependencies` to also build dependencies 
- Added source policies for build
- Included disabled services for shell completion
- Restored `Project` in `ps` json output
- Add OCI 1.0 fallback support for AWS ECR
- `build` do not require environment to be resolved
- Send out a cancel event on SIGINT/SIGTERM signal for `compose up`
- Fixed log by exposing services ports when `--verbose`
- Fixed configs are mounted under /\<id\>
- Fixed combination of `--pull always --no-build`
- Fixed race condition in log printer
- Fixed `docker compose up` teardown when command context is cancelled
```

For more detailed information, see the [release notes in the Compose repo](https://github.com/docker/compose/releases/).

## 2.24.0
{{< release-date date="2024-01-11" >}}

### Update
- Dependencies upgrade: bump docker cli v1.25 beta1 
- Dependencies upgrade: bump cli to 25.0.0-beta.3
- Dependencies upgrade: bump compose-go to 2.0.0-beta.3
- Dependencies upgrade: bump ddev to 1.22.4 
- Dependencies upgrade: bump golang to 1.21.6 

> - Dependencies upgrade: bump golangci-lint to 1.55.2

Reviewer: I'm not sure we need to mention the dependency bump of `ddev` and `golang-ci`, those are more linked to CI
```suggestion
- Dependencies upgrade: bump docker cli v1.25 beta1 
- Dependencies upgrade: bump cli to 25.0.0-beta.3
- Dependencies upgrade: bump compose-go to 2.0.0-beta.3
- Dependencies upgrade: bump golang to 1.21.6 
```

Author: removed in https://github.com/docker/docs/pull/19089/commits/4bdc7942d3aa6d5a8e9cc18c6019098908d09472
